source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,462 @@
+   on using :command:`git format-patch`
+
+   .. code-block:: bash
+      
+      git format-patch -<n>
+
+
+#. Copy the patch files from the patches directory in the linux 
+   source tree to the RPM build directory.
+
+   .. code-block:: bash
+      
+      cp *.patch ~/clearlinux/packages/linux/
+
+
+#. Navigate back to the RPM build directory.
+
+   .. code-block:: bash
+      
+      cd ~/clearlinux/packages/linux/
+
+
+#. Open the Linux kernel package RPM SPEC file in an editor.
+
+   .. code-block:: bash
+
+      $EDITOR linux.spec
+
+
+#. Locate the section of the SPEC file that contains existing patch 
+   variable definitions and append your patch file name. Ensure the 
+   patch number does not collide with an existing patch. 
+   In this example, the patch file is called 
+   :file:`2001-my-patch-for-driver-A.patch`
+
+   .. code-block:: bash
+
+      #
+      # Small Clear Linux Tweaks
+      #
+      Patch0501: 0501-zero-extra-registers.patch
+      Patch0502: 0502-locking-rwsem-spin-faster.patch
+
+      #Serie1.name WireGuard
+      #Serie1.git  https://git.zx2c4.com/WireGuard
+      #Serie1.tag  00bf4f8c8c0ec006633a48fd9ee746b30bb9df17
+      Patch1001: 1001-WireGuard-fast-modern-secure-kernel-VPN-tunnel.patch
+      #Serie1.end
+
+      Patch2001: 2001-my-patch-for-driver-A.patch
+
+
+#. Locate the section of the SPEC file further down that contains 
+   patch application and append your patch file number used in the step above.
+   In this example, patch2001 is added.
+
+   .. code-block:: bash
+
+      #
+      # Small tweaks
+      #
+      %patch0501 -p1
+      %patch0502 -p1
+
+      #Serie1.patch.start
+      %patch1001 -p1
+      #Serie1.patch.end
+
+      %patch2001 -p1
+
+
+#. Commit and save the changes to the RPM SPEC file.
+
+
+
+Modify kernel boot parameters
+=============================
+The kernel boot options are passed from the bootloader to the kernel with 
+command-line parameters. 
+
+While temporarily changes can be made to kernel parameters on a running 
+system or on a during boot, you can also modify the default parameters that 
+are persistent and distributed with a customized kernel.
+
+
+#. Open the kernel :file:`cmdline` file in an editor.
+
+   .. code-block:: bash
+
+      $EDITOR cmdline
+
+
+#. Make any desired change to the kernel parameters. 
+   For example, you can remove the :command:`quiet` parameter to see more 
+   verbose output of kernel log messages during the boot process.
+
+
+#. Commit and save the changes to the :file:`cmdline` file.
+
+See the 
+`Kernel parameters documentation`_ for a list of available parameters. 
+
+
+
+Build and install the kernel
+****************************
+After changes have been made to the kernel source and RPM SPEC file, 
+the kernel is ready to be compiled and packaged into an RPM.
+
+#. Start the compilation process by issuing the :command:`make build` command. 
+   This process is typically resource intensive and will take a while.
+
```

Comment:
Might be good to add a note about enabling ccache plugin for mock to speed up rebuilds.